builtin/docker/builder.go

```diff
@@ -424,14 +452,28 @@ func (b *Builder) buildWithDocker(
 		return status.Errorf(codes.InvalidArgument, "buildkit is required to use platform option")
 	}
 
+	var authMap = make(map[string]types.AuthConfig)
+	if (Auth{} != *b.config.Auth) {
```

Comment:
This is a really strange conditional, I guess just checking if the value is empty. I've never seen this before, maybe leave a comment (and personally I'd put the `Auth{}` on the right hand side).

Reply:
Agreed, I couldn't find a simpler way to check if a struct is empty, though! Will add a comment and move to right side.